config/main.py

```diff
@@ -726,6 +729,47 @@ def reload():
     else:
         click.secho('Buffer definition template not found at {}'.format(buffer_template_file), fg='yellow')
 
+#
+# 'snmp-community' command
+#
+@config.command('snmp-community')
+@click.argument('community-name', metavar='<community_name>', required=True)
+def snmp_community(community_name):
+    """Add SNMP community string"""
+
+    """Serialization of a Python dict from a SNMP YAML File """
+    with open(SNMP_COMMUNITY_FILE, 'r') as stream:
+        try:
+            snmp_community_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as error:
+            click.echo("SNMP config file not loaded correctly with error {}".format(error))
+            raise
+
+    if 'snmp_rocommunity' in snmp_community_dict.keys():
+        existing_community_name = snmp_community_dict['snmp_rocommunity']
```

Comment:
>existing_community_name = snmp_community_dict['snmp_rocommunity'] [](start = 8, length = 65)

You could move this line into the if-block.

Reply:
Still applicable.

---
In reply to: [339219893](https://github.com/Azure/sonic-utilities/pull/687#discussion_r339219893) [](ancestors = 339219893)